qiskit/transpiler/passmanager.py

```diff
@@ -131,16 +131,21 @@ def run_passes(self, dag):
 
     def run(self, circuit):
         """Run all the passes on a QuantumCircuit
+
         Args:
             circuit (QuantumCircuit): circuit to transform via all the registered passes
+
         Returns:
             QuantumCircuit: Transformed circuit.
         """
+        name = circuit.name
         dag = circuit_to_dag(circuit)
+        del circuit
         for passset in self.working_list:
```

Comment:
There is some redundancy here. Considering
```
-        for passset in self.working_list:
-            for pass_ in passset:
-                dag = self._do_pass(pass_, dag, passset.options)
-        circuit = dag_to_circuit(dag)
-        return circuit
-
+        return dag_to_circuit(self.run_passes(dag))
+        
```

Reply:
good catch thanks